task_processing/plugins/persistence/dynamodb_persistence.py

```diff
@@ -0,0 +1,75 @@
+import datetime
+import decimal
+
+import boto3.session as bsession
+from boto3.dynamodb.conditions import Key
+from pyrsistent import thaw
+
+from task_processing.interfaces.persistence import Persister
+
+
+class DynamoDBPersister(Persister):
+    def __init__(self, table_name, endpoint_url=None, session=None):
+        self.table_name = table_name
+        if not session:
+            session = bsession.Session()
+        self.ddb_client = session.client(
+            service_name='dynamodb',
+            endpoint_url=endpoint_url,
+        )
+        self.table = session.resource(
+            endpoint_url=endpoint_url,
+            service_name='dynamodb'
+        ).Table(table_name)
+
+    def read(self, task_id, comparison_operator='EQ'):
+        res = self.table.query(
+            KeyConditionExpression=Key('task_id').eq(task_id)
+        )
+        return [self._replace_decimals(item) for item in res['Items']]
+
+    def write(self, event):
+        return self.ddb_client.put_item(
+            TableName=self.table_name,
+            Item=self._event_to_item(event)
+        )
+
+    def _event_to_item(self, event):
+        thawed = thaw(event)
+        resp = {}
+        for k, v in thawed.items():
+            if type(v) is str:
+                resp[k] = {
+                    'S': v
+                }
+            elif type(v) is datetime.datetime:
+                resp[k] = {
+                    'N': str(int(v.timestamp()))
+                }
+            elif type(v) is bool:
+                resp[k] = {
+                    'S': str(v)
+                }
+            elif type(v) is int:
+                resp[k] = {
+                    'N': str(v)
+                }
+            elif type(v) is dict:
+                resp[k] = {
+                    'M': self._event_to_item(v)
+                }
+            elif type(v) is list:
```

Comment:
One thing I'd note is that using `type(x) is y` is kind of an antipattern in Python - I'd prefer `isinstance`, although I have no context as to where the data is coming from, so it might not make a difference.

In this specific case, you actually support any iterable, not just a list. I'd probably use something like `if isinstance(e, collections.Iterable):` here.